### ISSUE TITLE
Domains cache cannot be updated correctly when the server time changes

### ISSUE THREAD

**Issue by luozhiwenn** [miekg/coredns#5478]
**What happened**:
When the time of the server where the coredns resides change(set the time to a certain time in the past), the domains cache cannot be updated correctly.

**What you expected to happen**:
domains cache can be updated correctly
I think coredns should use [golang monotonic clock](https://golang.org/pkg/time/#hdr-Monotonic_Clocks) to calculate ttl in cache plugin

**How to reproduce it (as minimally and precisely as possible)**:
- run CoreDNS in kubernetes cluster as DNS server
- create a service `test-svc`
- initiate a domain name resolution request for this service, get the service ClusterIP
![image](https://user-images.githubusercontent.com/74550097/176848260-ca5e337d-5f78-425e-ab96-a291c6b4e330.png)
- use `date -s "xxx"` to change the time one hour back
- delete the created service `test-svc` then recreate, kubernete will allocate a new ClusterIP for this service
- initiate a domain name resolution request again but got a stale ClusterIP that first time created, even though the record ttl has passed
![image](https://user-images.githubusercontent.com/74550097/176848325-0982f163-3db1-410b-9350-08501146c567.png)

**Anything else we need to know?**:

**Environment**:

- the version of CoreDNS: v1.8.3
- Corefile: 
- logs, if applicable:
- OS (e.g: `cat /etc/os-release`): EulerOS
- Others:

**Comment by chrisohaver** [comment]
At a glance, it looks like _cache_ handles this OK.  Monotonic clock operations are handled implicitly by Golang.  Time.time contains a monotonic time and a local time (aka wall clock); operations such as `Sub` use the monotonic times if available.  I don't see anything that would strip the monotonic times.

A replication without the _kubernetes_ plugin would be more minimal.  Many confounding factors in a K8s environment, e.g.  time serialization between multiple k8s processes, multiple k8s nodes each with their own clock ...

**Comment by luozhiwenn** [comment]
Maybe the `now.UTC()` strip the monotonic times?
https://github.com/coredns/coredns/blob/c4d02442ef493b4892316fea00128c982d48a914/plugin/cache/item.go#L97

I found such a sentence in [golang monotonic clock](https://golang.org/pkg/time/#hdr-Monotonic_Clocks):
![image](https://user-images.githubusercontent.com/74550097/176995539-6cb5082c-a52a-45c6-a7e8-4bae423ac49d.png)
[](url)

**Comment by luozhiwenn** [comment]
I found such a sentence in [[golang monotonic clock](https://golang.org/pkg/time/#hdr-Monotonic_Clocks)](https://golang.org/pkg/time/#hdr-Monotonic_Clocks):

**Comment by chrisohaver** [comment]
Yes. Thanks, I didn’t see that initially. It is not mentioned in the UTC function Godoc, like it is for Round and Truncate.

**Comment by chrisohaver** [comment]
This affects record expiration time calculations in upstream miekg/dns as well.
I see manual calculations done in there (that use UTC and unixtime), which don't take advantage of the time.Time monotonic clock.
